### ISSUE TITLE
Implement threaded widget.event_generate() / TkApp.call()

### ISSUE THREAD

**Issue by James-E-A** [pypy/pypy#5066]
If [tkdocs\.org's article on the event loop](https://tkdocs.com/tutorial/eventloop.html) is to be believed, `widget.event_generate()` is **the** thread-safe way to interact with it:

> If you need to communicate from another thread to the thread running Tkinter, keep it as simple as possible. Use `event_generate` to post a virtual event to the Tkinter event queue, and then bind to that event in your code.

which means I guess there's a bit of a blocker for running multithreaded GUI programs on PyPy:

https://github.com/pypy/pypy/blob/release-pypy3.10-v7.3.17/lib_pypy/_tkinter/app.py#L310  
https://github.com/python/cpython/blob/v3.12.7/Modules/_tkinter.c#L1471

**Comment by mattip** [comment]
Just above that is a comment discouraging multi-threaded tk apps. Do you have an example of something that works with CPython and fails on PyPy?

**Comment by mattip** [comment]
Related to #1929

**Comment by mattip** [comment]
And #3821

**Comment by James-E-A** [comment]
> Do you have an example of something that works with CPython and fails on PyPy?

Yes, [here's](https://gist.github.com/ad3c746f8c12332dc26b428d611bd3bd) a simple example of that; any situation where you're doing communication from a worker or helper thread to the main loop is affected by this (unless you build a solution using polling or “out-of-band” socket wakeups, of course.)
